### PR TITLE
Connected AssetMeta Component

### PR DESCRIPTION
**Pull request by xitman78**
AssetMeta component was connected with SDK.
Added loading spinner.
Added tracking of 'assetId' prop changes.
Added unit test.

### CODE REVIEW

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

File: src/api/api.ts [hunk 1]
```diff
@@ -20,3 +20,46 @@ export async function getAssetList({
...
+  return [];
+}
+
+export async function getAssetFiles(query: {
+  assetId: number;
+  limit: number;
+}): Promise<File[]> {
+  const response = await Files.list(query);
+
+  if (response.items && response.items.length > 0) {
```
Comment (f1cognite) [inline src/api/api.ts]: I think `response.items` will always be an array. So think this if-statement is unnecessary

File: src/components/AssetEventsPanel/AssetEventsPanel.tsx [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+import { AssetEventsPanelProps } from '../../interfaces';
 import { momentFromTimestamp } from '../../utils';
 
-interface EventAddonsProp extends ApiEvent {
+interface EventAddonsProp extends Event {
   typeAndSubtype: React.ReactNode;
   description: string;
   start: string;
   end: string;
+  metadata?: { [key: string]: any };
```
Comment (f1cognite) [inline src/components/AssetEventsPanel/AssetEventsPanel.tsx]:
```suggestion
  metadata?: { [key: string]: string };
```

File: src/api/api.ts (hunk 1 above)
Comment (xitman78) [reply src/api/api.ts]: fixed

File: src/components/AssetEventsPanel/AssetEventsPanel.tsx (hunk 2 above)
Comment (xitman78) [reply src/components/AssetEventsPanel/AssetEventsPanel.tsx]: fixed

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
I would like to have a documentation page as well in the Storybook (same as for SensorOverlay) and some better mock data for Events (not having `***` for the event name but fake some names)

**Comment by xitman78** [comment]
@f1cognite 
> I would like to have a documentation page as well in the Storybook (same as for SensorOverlay) and some better mock data for Events (not having `***` for the event name but fake some names)

'***'  - this is event type and as far as I can see API also returns '***' for event IDs used in mocks. What is the possible values for this event.type ?

**Comment by f1cognite** [comment]
Yes, from publicdata er have anonymized the data and replaced it with ***. For event.type it is free text and can be whatever like "failure", "alert", "workorder". When it comes to IDs is has to be a 53 bit number starting from 0 (safe for JavaScript)

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

**Comment by f1cognite** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-132.gearboxjs.preview.cogniteapp.com

**Pull request merged**